examples/llm_compression/torch/qat_with_nls_downstream/main.py

```diff
@@ -0,0 +1,773 @@
+                best_config = median_lora_rank_config
+
+            # Test the most frequent configuration
+            most_frequent_lora_rank_config = get_most_frequent_config(activation_counter)
+            configure_lora_adapters(
+                layer_id_vs_lora_quantizers_map,
+                specific_rank_config=most_frequent_lora_rank_config,
+            )
+            results_of_nls_finetuned_compressed_model_most_frequent = lm_eval(
+                model, task=args.task, batch_size=args.eval_batch_size, tokenizer=tokenizer
+            )
+            print(
+                f"Results of quantization-aware-finetuned (NLS-Most-Frequent) NNCF compressed model="
+                f"{json.dumps(results_of_nls_finetuned_compressed_model_most_frequent, indent=4)}"
+            )
+            overall_result["nls_results"].append(
+                {
+                    "type": "most-frequent",
+                    "config": most_frequent_lora_rank_config,
+                    "results": results_of_nls_finetuned_compressed_model_most_frequent,
+                }
+            )
+            if results_of_nls_finetuned_compressed_model_most_frequent[args.lm_eval_metric] > best_result:
+                best_result = results_of_nls_finetuned_compressed_model_most_frequent[args.lm_eval_metric]
+                best_config = most_frequent_lora_rank_config
+
+            # Test the top 5 min loss configurations
+            top_5_min_loss_configs = get_top_k_min_loss_configs(loss_recorder, k=5)
+            for i, min_loss_config in enumerate(top_5_min_loss_configs):
+                configure_lora_adapters(
+                    layer_id_vs_lora_quantizers_map,
+                    specific_rank_config=min_loss_config,
+                )
+                results_of_nls_finetuned_compressed_model_min_loss = lm_eval(
+                    model, task=args.task, batch_size=args.eval_batch_size, tokenizer=tokenizer
+                )
+                print(
+                    f"Results of quantization-aware-finetuned (NLS-Min-Loss-{i + 1}) NNCF compressed model="
+                    f"{json.dumps(results_of_nls_finetuned_compressed_model_min_loss, indent=4)}"
+                )
+                overall_result["nls_results"].append(
+                    {
+                        "type": f"min-loss-{i + 1}",
+                        "config": min_loss_config,
+                        "results": results_of_nls_finetuned_compressed_model_min_loss,
+                    }
+                )
+                if results_of_nls_finetuned_compressed_model_min_loss[args.lm_eval_metric] > best_result:
+                    best_result = results_of_nls_finetuned_compressed_model_min_loss[args.lm_eval_metric]
+                    best_config = min_loss_config
+        else:
+            assert args.custom_rank_config is not None, "Please provide `custom_rank_config` for evaluation."
```

Comment:
This check should happen as early as possible, otherwise customer will waste some time in a default case

Reply:
@ljaljushkin Fixed. Thanks!